### DISCUSSION TITLE
Noob here, couple of questions

### DISCUSSION THREAD

**Discussion by 2k43l3aj5l** [seleniumbase/SeleniumBase#2335] [Q&A] [answered]
Hi, I'm trying to figure this out:
### 1:
I'm trying to slow scroll to a specific element, and because there are multiple different elements with the same criteria and how my code works, I decided to use a `for` loop and do it like this:
```
randomhold = sb.driver.find_elements("css selector", ".randomclass")
for i in randomhold:
 differentrandomhold = i.find_element("css selector", ".differentrandomclass")
 differentrandomhold.slow_scroll_to()`
```
But then, I get this error:
`AttributeError: 'WebElement' object has no attribute 'slow_scroll_to'`
### 2:
If no matching element is found when using `find_element`, my program closes and I get this error:
`selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":".differentrandomclass"}`
Instead of exiting the program and giving me that exception in the console, how can I handle it so that `if` that element is not found, it will go to `else` block instead?

**Comment by mdmintz** [comment, marked as answer]
Hello. The `WebElement` doesn't contain the `seleniumbase` methods (with a few exceptions for UC Mode).

Here's an example of `slow_scroll_to()`:

```python
from seleniumbase import SB

with SB() as sb:
    sb.open("https://xkcd.com/color/rgb/")
    sb.slow_scroll_to("center > table tbody tr:nth-child(36)")
    sb.sleep(1)
```

Use methods such as `is_element_visible(selector)` and `is_element_present(selector)` to handle flow-of-control. Eg:

```python
if sb.is_element_visible(selector):
    # Do something here
else:
    # Do something else here
```

You can also use `try/except` blocks for error-handling. Eg:

```python
try:
    # Do something here
except Exception:
    # Do something else here
```

**Reply by 2k43l3aj5l** [reply]
Can I slow scroll to the xpath extracted from the found element?

```
 differentrandomhold = i.find_element("css selector", ".differentrandomclass")
 xpath = differentrandomhold.get_attribute("xpath")
 print(xpath)
 sb.slow_scroll_to(xpath)
```
If I try to do it like that, printed xpath shows `"None"`, and then I get:
```
Exception: Invalid selector type: "<class 'NoneType'>"
Expecting a selector of type: "<class 'str'>" (string)!
```

**Reply by mdmintz** [reply]
`xpath` isn't an attribute of `WebElement`. Your call would return a `None` object, which you saved into your variable.

**Reply by mdmintz** [reply]
This is how `hover()` works:

```python
element = driver.find_element(by=by, value=selector)
hover = ActionChains(driver).move_to_element(element)
hover.perform()
```

Methods such as `move_to_element()` are part of `selenium`, not `seleniumbase`, so I don't have the specifics memorized.

But if you're trying to avoid detection, you shouldn't be using `hover()` at all, as that can trigger detection. There's a `uc_click()` method specifically for undetected clicking. See the example: [SeleniumBase/examples/raw_turnstile.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_turnstile.py)
